component/block_cache/block_cache.go

```diff
@@ -253,6 +254,23 @@ func (bc *BlockCache) Configure(_ bool) error {
 		bc.diskTimeout = conf.DiskTimeout
 	}
 
+	var noDu bool
+	if err := config.UnmarshalKey("no-du", &noDu); err == nil {
+		if noDu {
+			bc.diskUsageConfiguration = common.DiskUsageConfiguration{
+				DiskUsageFunction: common.GetUsageWithWalkInMegabytes,
+				UsesDu:            false,
+			}
+		} else {
+			bc.diskUsageConfiguration = common.DiskUsageConfiguration{
```

Comment:
Instead of being a file-cache or a block-cache member this could be a global variable in itself where GetUsage is generalised so that in any code space in blobfuse if we try to compute the disk usage it uses the configured method. By doing it in this format we are saying file-cache and block-cache will define du or no-du but other components are still free to rely on du command. Ideally a global function pointer shall be set based on the input cli flag and all other component shall just use that function pointer to get disk space, globally that function defines whether to use du or not.